### ISSUE TITLE
[1] does not extend [...any]

### ISSUE THREAD

**Issue by mtomassoli** [microsoft/TypeScript#44016]
# Bug Report

### 🕗 Version & Regression Information

Version 4.1.5: OK
Version 4.2.3: BUG
Version 4.3.0(beta): BUG

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.3.0-beta#code/C4TwDgpgBA6gogSQEoBEA8AVAfFAvFDKCAD2AgDsATAZygG0A6JgQ3JAF0oB+KYAJwCu0AFxQAZswA21CAG4AUKEgEAjHliJUaOuyyyoBw0eNQA9Kd6CIi8NAwAmdfGTo6K3fpNezF-kJvKGADMTpquKgA0UPYe3kbmlv4B0M6oavg6RKQUNPRMDKwc3Iki4lIynsYJftZKKWGOGe5ZZFS0jCxsnDw1UKIS0nImCQMyyRouIU1RMS057fmF3SV9ZYOV1VbyQA)

### 💻 Code

```ts
type WEIRD<T> = T extends [...any] ? true : false;
type T1 = WEIRD<[]>;                // true
type T2 = WEIRD<[1]>;               // true
type T3 = WEIRD<[1, 2]>;            // true

type WEIRD1 = [] extends [...any] ? true : false;       // true
type WEIRD2 = [1] extends [...any] ? true : false;      // false
type WEIRD3 = [1, 2] extends [...any] ? true : false;   // true
```

### 🙁 Actual behavior

WEIRD2 = false

### 🙂 Expected behavior

WEIRD2 = true

**Comment by MartinJohns** [comment]
https://www.typescriptlang.org/docs/handbook/2/conditional-types.html#distributive-conditional-types

You're looking for:

    type WEIRD2 = [[1]] extends [[...any]] ? true : false;      // true

**Comment by mtomassoli** [comment]
@MartinJohns Where's the union? And why does it work with v4.1.5 which also supports distributive conditional types?

**Comment by andrewbranch** [comment]
This is already fixed in 4.4 beta.
